plugin/evm/vm_test.go

```diff
@@ -2288,3 +2288,104 @@ func TestBuildAllowListActivationBlock(t *testing.T) {
 		t.Fatalf("Expected allow list status to be set to Admin: %s, but found: %s", precompile.AllowListAdmin, role)
 	}
 }
+
+// Test that the tx allow list allows whitelisted transactions and blocks non-whitelisted addresses
+func TestTxAllowListSuccessfulTx(t *testing.T) {
+	// Setup chain params
+	genesis := &core.Genesis{}
+	if err := genesis.UnmarshalJSON([]byte(genesisJSONSubnetEVM)); err != nil {
+		t.Fatal(err)
+	}
+	genesis.Config.TxAllowListConfig = precompile.TxAllowListConfig{
+		AllowListConfig: precompile.AllowListConfig{
+			BlockTimestamp:  big.NewInt(0),
+			AllowListAdmins: testEthAddrs[0:1],
+		},
+	}
+	genesisJSON, err := genesis.MarshalJSON()
+	if err != nil {
+		t.Fatal(err)
+	}
+	issuer, vm, _, _ := GenesisVM(t, true, string(genesisJSON), "", "")
+
+	defer func() {
+		if err := vm.Shutdown(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	newTxPoolHeadChan := make(chan core.NewTxPoolReorgEvent, 1)
+	vm.chain.GetTxPool().SubscribeNewReorgEvent(newTxPoolHeadChan)
+
+	genesisState, err := vm.chain.BlockChain().StateAt(vm.chain.GetGenesisBlock().Root())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that address 0 is whitelisted and address 1 is not
+	role := precompile.GetTxAllowListStatus(genesisState, testEthAddrs[1])
+	if role != precompile.AllowListNoRole {
+		t.Fatalf("Expected allow list status to be set to no role: %s, but found: %s", precompile.AllowListNoRole, role)
+	}
+	role = precompile.GetTxAllowListStatus(genesisState, testEthAddrs[0])
+	if role != precompile.AllowListAdmin {
+		t.Fatalf("Expected allow list status to be set to no role: %s, but found: %s", precompile.AllowListAdmin, role)
+	}
+
+	// Submit a successful transaction
+	tx := types.NewTransaction(uint64(0), testEthAddrs[0], new(big.Int).Mul(firstTxAmount, big.NewInt(3)), 21000, big.NewInt(testMinGasPrice*3), nil)
```

Comment:
Can we replace the amount argument in these txs with `big.NewInt(1)` instead of the more complicated multiplication? I think this is borrowing logic that as applicable from another test.

We should be able to do the same thing and replace `big.NewInt(testMinGasPrice*3)` with `big.NewInt(testMinGasPrice)`.

Reply:
done